operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakIngress.java

```diff
@@ -0,0 +1,116 @@
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.operator.v2alpha1;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.networking.v1.IngressBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.api.model.networking.v1.Ingress;
+import org.keycloak.operator.Constants;
+import org.keycloak.operator.OperatorManagedResource;
+import org.keycloak.operator.StatusUpdater;
+import org.keycloak.operator.v2alpha1.crds.Keycloak;
+import org.keycloak.operator.v2alpha1.crds.KeycloakStatusBuilder;
+
+import java.util.Optional;
+
+public class KeycloakIngress extends OperatorManagedResource implements StatusUpdater<KeycloakStatusBuilder> {
+
+    private Ingress existingIngress;
+    private final Keycloak keycloak;
+
+    public KeycloakIngress(KubernetesClient client, Keycloak keycloakCR) {
+        super(client, keycloakCR);
+        this.keycloak = keycloakCR;
+        this.existingIngress = fetchExistingIngress();
+    }
+
+    @Override
+    protected Optional<HasMetadata> getReconciledResource() {
+        if (existingIngress == null) {
+            return Optional.of(newIngress());
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    private Ingress newIngress() {
+        var port = (keycloak.getSpec().isHttp()) ? Constants.KEYCLOAK_HTTP_PORT : Constants.KEYCLOAK_HTTPS_PORT;
+        var backendProtocol = (keycloak.getSpec().isHttp()) ? "HTTP" : "HTTPS";
+
+        Ingress ingress = new IngressBuilder()
+                .withNewMetadata()
+                .withName(getName())
+                .withNamespace(getNamespace())
+                .addToAnnotations("nginx.ingress.kubernetes.io/backend-protocol", backendProtocol)
+                .endMetadata()
+                .withNewSpec()
+                .withNewDefaultBackend()
+                .withNewService()
+                .withName(keycloak.getMetadata().getName() + Constants.KEYCLOAK_SERVICE_SUFFIX)
+                .withNewPort()
+                .withNumber(port)
+                .endPort()
+                .endService()
+                .endDefaultBackend()
+                .addNewRule()
+                .withNewHttp()
+                .addNewPath()
+                .withPath("/")
+                .withPathType("ImplementationSpecific")
+                .withNewBackend()
```

Comment:
At best of my understanding this encoding is standard and follows the good practices

Reply:
Not sure. Even the most basic [examples](https://kubernetes.io/docs/concepts/services-networking/ingress/#single-service-ingress) state it just with default backend. But I'm not fussy on this, so let's keep it as is. 👍

Reply:
Agreed, just to leave more context, the old operator was using only the `path` without the `defaultBackend`.
K8s was rising a warning because of the missing default backend and I completed the resource with it.